hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/detailslayout/AbstractDistributionSetDetails.java

```diff
@@ -0,0 +1,214 @@
+/**
+ * Abstract class which contains common code for Distribution Set Details
+ *
+ */
+public abstract class AbstractDistributionSetDetails
+        extends AbstractNamedVersionedEntityTableDetailsLayout<DistributionSet> {
+
+    private static final long serialVersionUID = 1L;
+
+    private final DistributionAddUpdateWindowLayout distributionAddUpdateWindowLayout;
+
+    private final DistributionSetMetadatadetailsLayout dsMetadataTable;
+
+    private final UINotification uiNotification;
+
+    private final transient DistributionSetManagement distributionSetManagement;
+
+    private final DsMetadataPopupLayout dsMetadataPopupLayout;
+
+    private final DistributionTagToken distributionTagToken;
+
+    private final SoftwareModuleDetailsTable softwareModuleDetailsTable;
+
+    private VerticalLayout softwareModuleTab;
+
+    protected AbstractDistributionSetDetails(final VaadinMessageSource i18n, final UIEventBus eventBus,
+            final SpPermissionChecker permissionChecker, final ManagementUIState managementUIState,
+            final DistributionAddUpdateWindowLayout distributionAddUpdateWindowLayout,
+            final DistributionSetManagement distributionSetManagement,
+            final DsMetadataPopupLayout dsMetadataPopupLayout, final EntityFactory entityFactory,
+            final UINotification uiNotification, final TagManagement tagManagement,
+            final SoftwareModuleDetailsTable softwareModuleDetailsTable) {
+        super(i18n, eventBus, permissionChecker, managementUIState);
+        this.distributionAddUpdateWindowLayout = distributionAddUpdateWindowLayout;
+        this.uiNotification = uiNotification;
+        this.distributionSetManagement = distributionSetManagement;
+        this.dsMetadataPopupLayout = dsMetadataPopupLayout;
+        this.distributionTagToken = new DistributionTagToken(permissionChecker, i18n, uiNotification, eventBus,
+                managementUIState, tagManagement, distributionSetManagement);
+        this.softwareModuleDetailsTable = softwareModuleDetailsTable;
+
+        dsMetadataTable = new DistributionSetMetadatadetailsLayout(i18n, permissionChecker, distributionSetManagement,
+                dsMetadataPopupLayout, entityFactory, uiNotification);
+        createSoftwareModuleTab();
+    }
+
+    @Override
+    protected void onEdit(final ClickEvent event) {
+        final Window newDistWindow = distributionAddUpdateWindowLayout.getWindow(getSelectedBaseEntityId());
+        newDistWindow.setCaption(getI18n().getMessage(UIComponentIdProvider.DIST_UPDATE_CAPTION));
+        UI.getCurrent().addWindow(newDistWindow);
+        newDistWindow.setVisible(Boolean.TRUE);
+    }
+
+    @EventBusListenerMethod(scope = EventScope.UI)
+    void onEvent(final DistributionTableEvent distributionTableEvent) {
+        onBaseEntityEvent(distributionTableEvent);
+    }
+
+    @Override
+    protected String getDefaultCaption() {
+        return getI18n().getMessage("distribution.details.header");
```

Comment:
The UIComponentIdProvider is used for component's IDs. We don't have an id provider for message ids yet. It is a good point, we should introduce an id provider for message ids, but I would not do that in this pull request.